interfaces/sourcegen/sourcegen/_ochestrate.py

```diff
@@ -0,0 +1,54 @@
+# This file is part of Cantera. See License.txt in the top-level directory or
+# at https://cantera.org/license.txt for license and copyright information.
+
+import importlib
+import os
+import os.path
+import re
+
+import ruamel.yaml
+
+from ._types import *
+
+
+def _call_generate_source(generator, incl_file, ignore):
+    print('  ' + incl_file.path)
+
+    with open(incl_file.path, 'r') as f:
+        ct = f.read()
+
+    matches = re.finditer(r'CANTERA_CAPI.*?;', ct, re.DOTALL)
+    c_functions = [re.sub(r'\s+', ' ', m.group()) for m in matches]
+
+    if not c_functions:
+        return
+
+    parsed = (Func.parse(f) for f in c_functions)
+
+    if ignore:
+        print(f'    ignoring ' + str(ignore))
+
+    parsed = [f for f in parsed if f.name not in ignore]
+
+    generator.generate_source(incl_file, parsed)
+
+
+def generate_source(lang: str, out_dir: str):
+    print('Generating source files...')
+
+    my_path = os.path.dirname(__file__) + '/'
+    clib_path = os.path.normpath(my_path + '../../../include/cantera/clib')
+
+    with open(my_path + lang + '/config.yaml', 'r') as config_file:
+        config = ruamel.yaml.safe_load(config_file)
+
+    module = importlib.import_module(__package__  + '.' + lang)
+    generator = module.SourceGenerator(out_dir, config)
+
+    ignore = config['ignore']
+
+    for incl_file in os.scandir(clib_path):
+        if incl_file.name not in ignore or ignore[incl_file.name]:
+            _call_generate_source(generator, incl_file, ignore.get(incl_file.name, []))
```

Comment:
I can foresee the potential for language interfaces to want to be able to access information about more than one header file simultaneously. Do you think it could make sense to first generate the entire list of functions defined in the clib, filter it with the `ignore` list, and then pass that do `generator.generate_source`? This would also present the (minor) performance improvement of not needing to parse the entire set of header files on each build, if the information could be cached somehow.

I also wonder if it makes sense to allow language generators to dynamically filter the list from Python, rather than having a declarative configuration in YAML (or perhaps allow both). Maybe this filtering should be pushed down to the `SourceGenerator` instances?

Reply:
totally redone, new commits to be pushed shortly...